libs/@local/harpc/client/typescript/src/net/internal/transport.ts

```diff
@@ -100,46 +89,30 @@ const resolveDnsMultiaddrSegment = (code: number, value?: string) =>
     }
 
     const hostname = value;
+    const types: Dns.RecordType[] = [];
 
-    let family: 0 | 4 | 6;
-    if (code === DNS4_PROTOCOL.code) {
-      family = 4;
-
-      if (isIPv4(hostname)) {
-        return [[IPV4_PROTOCOL.code, hostname] as const];
-      }
-    } else if (code === DNS6_PROTOCOL.code) {
-      family = 6;
-
-      if (isIPv6(hostname)) {
-        return [[IPV6_PROTOCOL.code, hostname] as const];
-      }
-    } else {
-      family = 0;
+    if (code === DNS_PROTOCOL.code || code === DNS4_PROTOCOL.code) {
+      types.push("A");
+    }
 
-      const hostnameFamily = isIP(hostname);
-      if (hostnameFamily === 4) {
-        return [[IPV4_PROTOCOL.code, hostname] as const];
-      } else if (hostnameFamily === 6) {
-        return [[IPV6_PROTOCOL.code, hostname] as const];
-      }
+    if (code === DNS_PROTOCOL.code || code === DNS6_PROTOCOL.code) {
+      types.push("AAAA");
     }
 
-    const responses = yield* Effect.tryPromise({
-      try: () => dns.lookup(hostname, { all: true, family }),
-      catch: (cause) => new DnsError({ cause }),
+    const records = yield* Dns.resolve(hostname, {
```

Comment:
I guess we're going to add `dns.resolveSoa` or use `dns.resolveAny` in a follow up?

Reply:
if any it's going to be `dns.resolveAny`, I didn't use `dns.resolve` because you can't get the TTL that way, you can with `resolve4`/`resolve6`. 

(I actually forgot about `dns.resolveAny`, I will add it if this works! 🤞)

Reply:
Yup, let's see if it's working in the first place. If `resolveAny` does not work we can explore `resolve4` or `resolve6` to get the TTL. (Or always try `resolve6` first and if that fails use `resolve4` as fallback)

Reply:
We already do that 😄 

https://github.com/hashintel/hash/blob/9576fa6e5e76b241e163cf088456986cc47844be/libs/%40local/harpc/client/typescript/src/net/internal/dns.ts#L38-L70

Reply:
We should test at some point what is faster, two dns calls as we have currently, or a single `resolveAny` which potentially unused data.